packages/trader/src/Modules/Trading/Components/Form/TradeParams/amount-mobile.tsx

```diff
@@ -17,28 +35,31 @@ const Basis = observer(
         selected_basis,
         setSelectedAmount,
         setAmountError,
-    }) => {
+    }: TBasis) => {
         const { ui, client } = useStore();
         const { addToast, vanilla_trade_type } = ui;
         const { currency } = client;
         const {
             onChangeMultiple,
-            trade_amount,
-            trade_basis,
-            trade_duration_unit,
-            trade_duration,
+            amount: trade_amount,
+            basis: trade_basis,
+            duration_unit: trade_duration_unit,
+            duration: trade_duration,
             contract_type,
             stake_boundary,
         } = useTraderStore();
+
         const user_currency_decimal_places = getDecimalPlaces(currency);
-        const onNumberChange = num => {
+        const onNumberChange = (num: number | string) => {
             setSelectedAmount(basis, num);
             validateAmount(num);
         };
-        const formatAmount = value =>
-            !isNaN(value) && value !== '' ? Number(value).toFixed(user_currency_decimal_places) : value;
-        const setBasisAndAmount = amount => {
-            const on_change_obj = {};
+        const formatAmount = (value: number | string) => {
+            const numericValue = typeof value === 'string' ? Number(value) : value;
+            return !isNaN(numericValue) ? parseFloat(numericValue.toFixed(user_currency_decimal_places)) : numericValue;
```

Comment:
```suggestion
        const formatAmount = (value: number | string) =>
            !isNaN(+value) && value !== '' ? Number(value).toFixed(user_currency_decimal_places) : value;
```
@henry-deriv please don't change the function logic because it doesn't work in the same way.
In the screenshot below, you can see that `formatAmount` (old function) works differently from `formatAmount2` which is `formatAmount` with your new logic:
<img width="562" alt="Screenshot 2023-07-27 at 3 03 25 PM" src="https://github.com/binary-com/deriv-app/assets/103177211/9db46060-07ab-4b2f-8c3b-8d37c56b0a98">

Reply:
done